core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -198,9 +198,23 @@ public boolean isReady() {
   /**
    * Enables per-message compression, if an encoding type has been negotiated.  If no message
    * encoding has been negotiated, this is a no-op.
+   *
+   * @param enabled whether or not subsequent calls to {@link #sendMessage} should be compressed.
    */
   @ExperimentalApi
   public void setMessageCompression(boolean enabled) {
-    // noop
+    throw new UnsupportedOperationException("unimplemented");
```

Comment:
Java doc seems best.

Really the methods should be abstract, but since ClientCall has to be stable this is the next best thing.  A really frustrating thing is to call a method and it do nothing.  This was changed to force those who override client call to do something with it.  Whether or not people actually do that, I don't know, but this would make it fail loudly rather than quietly.